### ISSUE TITLE
DataStore: support pausing/resuming of cloud sync

### ISSUE THREAD

**Issue by nylki** [aws-amplify/amplify-js#5037]
**Is your feature request related to a problem? Please describe.**
If I understand correctly, the DataStore currently always syncs updates immediately to the cloud. This may not be desirable for battery conservation and data usage.

**Describe the solution you'd like**
I would like the DataStore-API to support a manual cloud-sync functionality, so that the App could for example sync only a few times a day eg. when the device is connected to power.

eg.:

 ```.js
DataStore.sync()
```

or an option to pause syncing:

```.js
// When power is low
DataStore.pauseAutoSync()

// user has plugged in power
DataStore.resumeAutoSync()
```

and an optional config similiar to `fullSyncInterval`:

 ```.js
DataStore.configure({
    syncInterval: 60 * 12,
   fullSyncInterval: 60 * 24
})
```

 
**Describe alternatives you've considered**
Not using DataStore, but fetching/sending data  (eg. via regular graphQL) and store the Data manually in a storage-api (localstorage, indexedDB and equivalents on non-browsers platforms).

**Additional context**
If this should be already possible in some way, the documentation should be edited to include the feature.

**Comment by mdoesburg** [comment]
I need this feature too. Use case:

I am working on an app in which people might be in remote locations (in the jungle, top of a mountain etc.). In the app users have to be able to set if they only want to sync if they are for example on a WiFi network to prevent data charges.

**Comment by iqaldebaran** [comment]
I need this feature too¡¡¡
Use Case:
I am working in an app that have a free function with Datastore locally, and a pay function with cloud storage, and this feature proposed by @nylki is so useful for me.

**Comment by JPVan** [comment]
I would also love this feature. Both for a free vs paid user scenario, and also to have control over the number of writes to AppSync. In my use case, users edit only their own data, and only view data from other users. This data needs to be updated in the cloud only once per day, but will be entered locally several times a day. It would be nice to save to the cloud once per session, or only once per day.

**Comment by imoby** [comment]
Please add this!

**Comment by jtich** [comment]
+1

**Comment by jacobsapps** [comment]
Bumping this - it's needed badly. 

I've got a pre-sign-up flow where users create a bunch of data, which saves locally on a User object without an ID. When signing up, this is copied as a real User object with a UUID, and deleting the placeholder, but sometimes the placeholder syncs to DynamoDB too...

**Comment by pawelsas** [comment]
+1

**Comment by fizan2904** [comment]
+1

**Comment by sallymoon10** [comment]
+1

**Comment by nemezisSherokee** [comment]
Hello is the feature now available ?

**Comment by sacrampton** [comment]
Our use case is for highly intermittent networks - if its on/off/on/off/on.... then the sync is trying to restart over and over again and consumes essentially all of the phone's resources. This leads the app to hang for several minutes and in some cases eventually crash. Our customers are in very remote places that have highly intermittent networks. They are industrial sites that have some network access, but it varies as they walk around the site.

We need to be able to tell DataStore not to try to sync - alternatively, but be able to have it as a lower priority background process that does not overwhelm the application.

**Comment by emilyahsu** [comment]
Hi all, 

I'm Emily, a UX Designer on the AWS Amplify team, and we're conducting user interviews to learn more about the use cases of offline-first vs online-only development modes. The interview will be take at most 30 minutes, and as a “thank you”, we’d love to provide you with $25 in AWS credits for your time. 

If you are interested in participating, please DM me and we can schedule a time. Thanks!

**Comment by jreyesgfi** [comment]
Is there any update regarding this feature?

**Comment by sacrampton** [comment]
Please remove Stephen Crampton from your mailing list, He passed away this
year & I (his wife Megan's) don't have access to his passwords on this
platform/website,

Kind regards

Megan
